### DISCUSSION TITLE
Fetcher Improvements

### DISCUSSION THREAD

**Discussion by adityapurwa** [kawalcovid19/wargabantuwarga.com#139] [General]
Fetcher Improvements

# Current State

Benchmark:

> Using `time` command with no further prioritization

- Fetch Docs: 0.52s
- Fetch Sheets: 39.98s 🐌 
- Fetch FAQ: 0.38s

So the bottleneck is on the fetch sheets.

# Todo

## See if we can improve the current performance of the nodejs implementation ⌛ 

- Accessing the sheet file takes about 1-5s depending on the internet connection

**Comment by zainfathoni** [comment]
There's a potential solution at #137, have you taken a look at it @adityapurwa?

**Reply by adityapurwa** [reply]
Yes, however, I am trying to improve it locally in the meantime at https://github.com/kawalcovid19/wargabantuwarga.com/pull/141.

If #137 could provide us the JSON, I think during the build, we can just curl the JSON and save it to the data folder without further parsing needed.

**Reply by zainfathoni** [reply]
I see, thanks for taking a look at it!
